pilot/pkg/model/push_context.go

```diff
@@ -520,9 +530,13 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 	}
 
 	// if no public/private rule in calling proxy's namespace matched, and no public rule in the
-	// target service's namespace matched, search for any public destination rule across all namespaces
-	if host, ok := MostSpecificHostMatch(service.Hostname, ps.allExportedDestRules.hosts); ok {
-		return ps.allExportedDestRules.destRule[host].config
+	// target service's namespace matched, search for any public destination rule in the config root namespace
+	// NOTE: This does mean that we are effectively ignoring private dest rules in the config root namespace
```

Comment:
Probably not very likely. Maybe it should produce a error or warning at config time?

Reply:
I am not sure if the validation code will have access to the mesh config to detect and avoid this..